classes/requests/psr/class-truelayer-http-client.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * A PSR-18 compliant HTTP client that wraps the WordPress HTTP API.
+ *
+ * @package TrueLayer_For_WooCommerce/classes/requests/psr
+ */
+
+use KrokedilTrueLayerDeps\Nyholm\Psr7\Response;
+use KrokedilTrueLayerDeps\Psr\Http\Client\ClientInterface;
+use KrokedilTrueLayerDeps\Psr\Http\Message\RequestInterface;
+use KrokedilTrueLayerDeps\Psr\Http\Message\ResponseInterface;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class TrueLayer_Http_Client
+ */
+class TrueLayer_Http_Client implements ClientInterface {
+	/**
+	 * The log title to use for logging.
+	 *
+	 * @var string
+	 */
+	protected $log_title;
+
+	/**
+	 * The payment ID to use for the logging.
+	 *
+	 * @var string|null
+	 */
+	protected $payment_id;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param string      $log_title The log title to use for logging.
+	 * @param string|null $payment_id The payment ID to use for the logging.
+	 *
+	 * @return void
+	 */
+	public function __construct( $log_title = 'TrueLayer Request', $payment_id = null ) {
+		$this->log_title  = $log_title;
+		$this->payment_id = $payment_id;
+	}
+
+	/**
+	 * Sends a PSR-7 request and returns a PSR-7 response.
+	 *
+	 * @param RequestInterface $request The request to send.
+	 *
+	 * @return ResponseInterface
+	 */
+	public function sendRequest( RequestInterface $request ): ResponseInterface {
+		$uri          = $request->getUri();
+		$args         = $this->formatArgs( $request );
+		$http_version = $request->getProtocolVersion();
+
+		// Make the request.
+		$response = wp_remote_request( $uri, $args );
+
+		// Create the PSR-7 response.
+		$response_code    = wp_remote_retrieve_response_code( $response );
+		$response_headers = wp_remote_retrieve_headers( $response );
+		$response_reason  = wp_remote_retrieve_response_message( $response );
+		$response_body    = wp_remote_retrieve_body( $response );
+
+		// Ensure the response code is numeric.
+		$response_code = is_numeric( $response_code ) ? (int) $response_code : 400;
+
+		// Format the headers to ensure they are an array.
+		$response_headers = is_array( $response_headers ) ? $response_headers : iterator_to_array( $response_headers );
+
+		// Create the PSR-7 response.
+		$response = new Response( $response_code, $response_headers, $response_body, $http_version, $response_reason );
+
+		$this->logRequest( $uri, $args, $response_body, $response_code );
```

Comment:
Lets remove the response body from this log as it may end up logging PII data.

Reply:
We generally log the response body for requests to be able to see things more easily when dealing with support tickets. But we can remove it if needed. I can take a look at changing this in that case.